Some new Mac keyboards have two keycodes swapped.
It seems that most modern Mac have these keys swapped,
so make it the default and see who complains.

Not submitted yet, but partially discussed in
  https://bugs.freedesktop.org/show_bug.cgi?id=8068

Index: xkb-data/rules/base.m_k.part
===================================================================
--- xkb-data.orig/rules/base.m_k.part
+++ xkb-data/rules/base.m_k.part
@@ -3,7 +3,9 @@
   sun4		=	sun(type4_euro)
   sun5		=	sun(type5_euro)
   macintosh_old	=	macintosh(old)
-  macintosh	=	macintosh
+  macintosh_old2	=	macintosh
+  macbook	=	macintosh(macbook)
+  macintosh	=	macintosh(macbook)
   powerpcps2	=	powerpcps2
   pc98		=	xfree98(pc98)
   abnt2		=	xfree86(abnt2)
Index: xkb-data/rules/base.lists.part
===================================================================
--- xkb-data.orig/rules/base.lists.part
+++ xkb-data/rules/base.lists.part
@@ -15,7 +15,7 @@
 ! $pcgeometries = abnt2 latitude jp106 kr106
 
 ! $maclaptop = ibook powerbook
-! $macs = macintosh macintosh_old ibook powerbook
+! $macs = macintosh macintosh_old macintosh_old2 ibook powerbook macbook
 
 ! $macvendorlayouts = ch de dk es fi fr gb is it nl no pt se us
 
